scrabble.js

```diff
@@ -1,8 +1,142 @@
 var Scrabble = function() {};
+Scrabble.prototype.score = function(word) {
+  var letterScore = {"A": 1, "B": 3, "C": 3, "D": 2, "E": 1, "F": 4, "G": 2, "H": 4, "I": 1, "J": 8, "K": 5, "L": 1, "M": 3, "N": 1, "O": 1, "P": 3, "Q": 10, "R": 1, "S": 1, "T": 1, "U": 1, "V": 4, "W": 4, "X": 8, "Y": 4, "Z": 10};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+  word = word.toUpperCase();
+
+  this.word = word.split("");
+  var score = 0;
+  if (this.word.length == 7) {
+    score = 50;
+  }
+  this.word.forEach(function(i) {
+    score += letterScore[i];
+  });
+  return score;
+};
+
+Scrabble.prototype.highestScoreFrom = function(arrayOfWords)
+{
+  this.arrayOfWords = arrayOfWords;
+  var scoredWords = [];
+  var highestScore = 0;
+
+  for (var i = 0; i < arrayOfWords.length; i++) {
+    var word = arrayOfWords[i];
+    var scoredInt = this.score(word);
+    if (highestScore <= scoredInt){
+      highestScore = scoredInt;
+    }
+  }
+
+  for (var i = 0; i < arrayOfWords.length; i++) {
+    var word = arrayOfWords[i];
+    var scoredInt = this.score(word);
+    if(scoredInt == highestScore){
+      scoredWords.push(word);
+    }
+  }
+
+  if(scoredWords.length > 1)
+  {
+    scoredWords.sort(function(a, b ) {
+      return b.length - a.length;
+    });
+    if ((scoredWords[0].length)==7) {
+      return scoredWords[0];
+    } else{
+      return(scoredWords[scoredWords.length - 1]);
+    }
+  }
+  else{
+    return scoredWords[0];
+  }
+};
+
+var Player = function(name){
+  this.name = name;
+  this.plays =[];
+  this.scrabble = new Scrabble();
+};
+
+Player.prototype.play = function(word) {
+  if (this.hasWon()===false){
+    this.plays.push(word);
+  }
+  else {
+    return false;
+  }
+};
+
+Player.prototype.totalScore = function() {
+  var totalScore = 0;
+  for(var i = 0; i < this.plays.length; i++){
+    totalScore += this.scrabble.score(this.plays[i]);
+    console.log(this.scrabble.score(this.plays[i]));
```

Comment:
This is good for debugging but it results in duplicating the score logic multiple times. I'd recommend assigning this value to a variable and then printing it our for performance purposes.